staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -116,14 +115,27 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 
 		defaultGVK := scope.Kind
 		original := r.New()
-		trace.Step("About to convert to expected version")
-		obj, gvk, err := decoder.Decode(body, &defaultGVK, original)
+
+		validationDirective, err := fieldValidation(req)
 		if err != nil {
-			err = transformDecodeError(scope.Typer, err, original, gvk, body)
 			scope.err(err, w, req)
 			return
 		}
 
+		decoder := scope.Serializer.DecoderToVersion(s.Serializer, scope.HubGroupVersion)
+		trace.Step("About to convert to expected version")
+		obj, gvk, err := decoder.Decode(body, &defaultGVK, original)
+		if err != nil {
+			if !runtime.IsStrictDecodingError(err) || validationDirective == runtime.StrictFieldValidation {
+				err = transformDecodeError(scope.Typer, err, original, gvk, body)
+				scope.err(err, w, req)
+				return
+			}
+			if validationDirective == runtime.WarnFieldValidation {
+				warning.AddWarning(req.Context(), "", err.Error())
```

Comment:
I would have expected one warning per unknown field, this looks like it would build a very large warning